core/lib/zksync_core/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -130,7 +129,7 @@ impl<E: EthInterface> L1GasPriceProvider for GasAdjuster<E> {
 
     fn estimate_effective_pubdata_price(&self) -> u64 {
         // For now, pubdata is only sent via calldata, so its price is pegged to the L1 gas price.
-        self.estimate_effective_gas_price() * L1_GAS_PER_PUBDATA_BYTE as u64
+        self.estimate_effective_gas_price() * self.config.l1_gas_per_pubdata_byte
```

Comment:
The `L1_GAS_PER_PUBDATA_BYTE` variable is used in many other parts of the code, so we risk inconsistency (if config value changes, this component will consider a different value than others). Additionally, it doesn't feel right that in config when we set the mode (rollup/validium) we also need to set this gas per pubdata byte to zero. 


What if we have a trait `GasAdjuster` with all its logic identical to this file, and two structs that implement it:
```
struct RollupGasAdjuster {
    l1_gas_per_pubdata_byte: L1_GAS_PER_PUBDATA_BYTE
}

struct ValidiumGasAdjuster {
    l1_gas_per_pubdata_byte: 0
}

```

This also allows for different DA modes to have slightly different behaviour when predicting L1/DA cost of a transaction

Reply:
Please also consider this comment for other places where you add/change configs. Would the same reasoning apply?

Reply:
> The `L1_GAS_PER_PUBDATA_BYTE` variable is used in many other parts of the code, so we risk inconsistency (if config value changes, this component will consider a different value than others). Additionally, it doesn't feel right that in config when we set the mode (rollup/validium) we also need to set this gas per pubdata byte to zero.
>
> What if we have a trait `GasAdjuster` with all its logic identical to this file, and two structs that implement it:
> 
> ```
> struct RollupGasAdjuster {
>     l1_gas_per_pubdata_byte: L1_GAS_PER_PUBDATA_BYTE
> }
> 
> struct ValidiumGasAdjuster {
>     l1_gas_per_pubdata_byte: 0
> }
> ```
> 
> This also allows for different DA modes to have slightly different behaviour when predicting L1/DA cost of a transaction

I completely agree with this (actually, this design passed through my mind but I was not sure if it'd be desired), we'll move forward with this request.

> Please also consider this comment for other places where you add/change configs. Would the same reasoning apply?

For sure! We'll take a look at the configs we've added/changed and take a similar approach for that (in the case the conclusion is that it'd be right for us as it is, I'll also let you know so you can see if it's ok or if you have some other idea).

Reply:
> The `L1_GAS_PER_PUBDATA_BYTE` variable is used in many other parts of the code, so we risk inconsistency (if config value changes, this component will consider a different value than others). Additionally, it doesn't feel right that in config when we set the mode (rollup/validium) we also need to set this gas per pubdata byte to zero.
> 
> What if we have a trait `GasAdjuster` with all its logic identical to this file, and two structs that implement it:
> 
> ```
> struct RollupGasAdjuster {
>     l1_gas_per_pubdata_byte: L1_GAS_PER_PUBDATA_BYTE
> }
> 
> struct ValidiumGasAdjuster {
>     l1_gas_per_pubdata_byte: 0
> }
> ```
> 
> This also allows for different DA modes to have slightly different behaviour when predicting L1/DA cost of a transaction

A note on this, making this change will imply changing the actual `[eth_sender.gas_adjuster]` section for `[eth_sender.rollup_gas_adjuster]` and adding a `[eth_sender.validium_gas_adjuster]` one (in addition with another `match` on the `l1_bach_commit_data_generator_mode` config to instantiate either `RollupGasAdjuster` or `ValidiumGasAdjuster` (I think that now that this config is being used in more than one case we should think in another name for it, if you have any ideas on generalizing Rollup and Validium modes it'd be helpful).

Reply:
> A note on this, making this change will imply changing the actual [eth_sender.gas_adjuster] section for [eth_sender.rollup_gas_adjuster] and adding a [eth_sender.validium_gas_adjuster] one

Hmmm why is it needed though?

Reply:
> with another match on the l1_bach_commit_data_generator_mode config to instantiate either RollupGasAdjuster or ValidiumGasAdjuster

Yes, this will be required

Reply:
> > A note on this, making this change will imply changing the actual [eth_sender.gas_adjuster] section for [eth_sender.rollup_gas_adjuster] and adding a [eth_sender.validium_gas_adjuster] one
> 
> Hmmm why is it needed though?

Now that I think about it again, I'm 99% sure this won't be needed (I'll let you know otherwise with a proper explanation).

Reply:
I'd also suggest setting another constant/config for the Validium's `l1_pubdata_price` given that as said here: https://github.com/matter-labs/zksync-era/blob/2fa6a7369b0f4fbd94e0dda3cec06ceeda090d15/docs/guides/advanced/fee_model.md#validium--data-availability-configurations
> If you're running an alternative DA, you should adjust the l1_pubdata_price to roughly cover the cost of writing one byte to the DA, and set

Reply:
> I'd also suggest setting another constant/config for the Validium's l1_pubdata_price given that as said here:

ah, makes sense. It's indeed possible, but given we don't have alternative DAs (and we can run the first testnet version without one I think), we can add this config later

Reply:
As an addition to the above comment, feel free to comment and ask any question on the linked PR.